OSSMDOC-735: Service Mesh 2.3.1 Z Stream Release Notes (#54610)
openshift/openshift-docs: modules/ossm-rn-deprecated-features.adoc

Removed functionality no longer exists in the product.

== Deprecated and removed features {SMProductName} 2.3.1

> The `concurrency` field of the `spec.proxy` API is deprecated as of release 2.3.1, and will be removed in a future release. While the field is still supported in release 2.3.1, customers should start using the `concurrency` field of the `ProxyConfig` API.

Reviewer: ```suggestion
The `concurrency` field of the `spec.proxy` API is deprecated as of release 2.3.1, and is planned for removal in a future release. Although the field is still supported in release 2.3.1, begin using the `concurrency` field of the `ProxyConfig` API instead.
```

Author: done

> The `concurrency` field of the `spec.proxy` API is deprecated as of release 2.3.1, and is planned for removal in a future release. Although the field is still supported in release 2.3.1, customers should begin using the `concurrency` field of the `ProxyConfig` API.

Reviewer 2: Could we add this item to "known issues" and:
1. Explain how to workaround it with `ProxyConfig`.
2. Say that we are going to deprecate this config option in the future and recommend using `ProxyConfig`?

Author: Done

> The `concurrency` field specifies the number of worker threads to run. If the field is set to 0, then the number of worker threads available is equal to the number of CPU cores. If the field is not set, then the number of worker threads available defaults to 2.

Reviewer: ```suggestion
The `concurrency` field specifies the number of worker threads to run. If the field is set to `0`, then the number of worker threads available is equal to the number of CPU cores. If the field is not set, then the number of worker threads available defaults to `2`.
```

> In the following example, the `concurrency` field is set to 0.

Reviewer: ```suggestion
In the following example, the `concurrency` field is set to `0`.
```

Author: done. I often forget to tag these, so thanks for the reminder @rolfedh!

> In the following example, the `concurrency` field is set to `0`.

Reviewer 2: Instead of explaining what this is, I would say that the following ProxyConfig is the equivalent of `spec.proxy.concurrency` in the SMCP.

Author: @jewertow As a documentation practice, it's good to explain how to use something, so I'd like to keep this information. Some customers who want to do this might not already be familiar with spec.proxy.concurrency so saying they work the same might not help customers just learning to do this.

I also think you're bringing up a point I hadn't thought of - they basically work the same. So, maybe saying that provides value for customers already familiar with spec.proxy.concurrency.

I'll add a sentence saying they are equivalent prior to the explanation.

[source,yaml]
----
apiVersion: networking.istio.io/v1beta1
kind: ProxyConfig
metadata:

>   name: per-workload-proxyconfig

Reviewer 2: ```suggestion
  name: mesh-wide-concurrency
```

Author: done

>   namespace: example

Reviewer 2: ```suggestion
  namespace: <istiod-namespace>
```

Removed functionality no longer exists in the product.

== Deprecated and removed features {SMProductName} 2.3.1

The `concurrency` field of the `spec.proxy` API is deprecated as of release 2.3.1, and will be removed in a future release. While the field is still supported in release 2.3.1, customers should start using the `concurrency` field of the `ProxyConfig` API.

The `concurrency` field specifies the number of worker threads to run. If the field is set to 0, then the number of worker threads available is equal to the number of CPU cores. If the field is not set, then the number of worker threads available defaults to 2.

In the following example, the `concurrency` field is set to 0.

[source,yaml]
----
apiVersion: networking.istio.io/v1beta1
kind: ProxyConfig
metadata:
  name: per-workload-proxyconfig
  namespace: example
spec:

>   selector:

Reviewer 3: let's remove the whole `selector` block from here for now. The deprecated setting was global, so the example should also apply globally - we could maybe mention that users can use the selectors to restrict it to a specific set of workloads, but it's not necessarily required

Author: done

    matchLabels:
      app: ratings
  concurrency: 0

>   ----

Reviewer 3: I think the whitespace before this line (----) breaks the formatting

Author: Thanks for noticing this!
